File: 0739 daily-temperatures/solution.cs
public class Solution {
    public int[] DailyTemperatures(int[] t) {
        Stack<int> st= new();
       int[] res = new int[t.Length];
       for(int i=0;i<t.Length;i++) {
        while(st.Count>0 && t[st.Peek()]<t[i])
            res[st.Peek()]=i-st.Pop();
        st.Push(i);
       }
         return res;
    }
}

File: 0739 daily-temperatures/solution.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& t) {
        stack<int> st;
        vector<int> ans(t.size(), 0);
        for(int i=0;i<t.size();i++) {
            while(!st.empty() && t[st.top()] < t[i]) {
                ans[st.top()] = i-st.top();
                st.pop();
            }
            st.push(i);
        }
        return ans;
    }
};
